### DISCUSSION TITLE
colordiff could have special rules for IDs and numeric/date fields

### DISCUSSION THREAD

**Discussion by RollingStar** [beetbox/beets#3919] [Ideas]
Today the importer gave me a new MBid for an album. It's just a UUID like 39e73e7d-a25a-43b7-aa81-0035e9478950, unreadable. The string rules for the highlighter will still color same characters the same color.

Great for strings, not for IDs. I propose a simple
[color 1] -> [color 2] for IDs. Even when an "e" is in ID 1 and ID 2. Color them differently.

For numeric and date fields, we could have highlight rules for [new value is lower] and [new value is higher].

https://github.com/beetbox/beets/pull/3721/files#diff-9e6736f74918c0608860bf8ce0d4d9b3333adaa2025f7b7b4d28189cf64c1958R653

https://github.com/beetbox/beets/blob/e1a901dce5c7b867a5c84a438044c4093bcbed05/beets/ui/__init__.py#L552

**Comment by sampsyo** [comment]
That's a good idea! Highlighting for numbers and dates could get complicated, but at the very least not attempting a "string diff" on IDs seems like a good idea. :smiley: Would you mind opening a feature request issue?

**Reply by RollingStar** [reply]
> Highlighting for numbers and dates could get complicated

Do you have an example? In my mind it would be a 'whitelist' of non-string fields. Catalogs numbers/descriptors would probably stay normal strings, even if entirely numeric.

For date comparison, there are probably libraries that do it, but I know you like keeping the dependency list small. Non-MB-matched tags could have dates in any format, but we could skip those or apply a best guess (using a date library). Worst case a smaller date is actually printed in the "larger date" color, which isn't much different from present state.
